src/lib/hardware/pci.lua

```diff
@@ -163,7 +163,7 @@ end
 --
 -- example: qualified("01:00.0") -> "0000:01:00.0"
 function qualified (address)
-   return address:gsub("^%d%d:%d%d[.]%d+$", "0000:%1")
+   return address:gsub("^[0-9a-f][0-9a-f]:[0-9a-f][0-9a-f][.][0-9a-f]+$", "0000:%1")
```

Comment:
Good catch! In Lua there's wildcard '%x' which matches hexadecimal digits. See http://www.lua.org/pil/20.2.html.